File: backend/src/WorkflowHub.Common/Responses/ApiResponse.cs
namespace WorkflowHub.Common.Responses;

/// <summary>
/// Standard success response envelope: { success, data, error }.
/// See ai-rules/backend/02-api/api-response-contract-rules.md.
/// </summary>
public class ApiResponse<T>
{
    public bool Success { get; set; }

    public T? Data { get; set; }

    public ApiError? Error { get; set; }
}

File: frontend/src/app/shared/types/api/api-response.ts
/**
 * Centralized transport contract types shared across all features.
 * Mirrors the backend api-response-contract.
 */

export interface ApiResponse<T> {
  success: boolean;
  data: T | null;
  error: ApiError | null;
}

export interface ApiError {
  code: string;
  message: string;
}

export interface PagedResult<T> {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
}
